female_lasso_mediation.rmd

```diff
@@ -0,0 +1,218 @@
+
+
+#Prepare dataset 
+#Select relevanat columns
+relevant_wave6_data <- lsac_wave6 %>% 
+  select(hicid, fcnfsad2:fcnfseo2d)
+
+relevant_wave8_data <- lsac_wave8 %>% 
+  select(hicid, hapsoc:hasdqtb)
+
+#Join datasets
+domsp_wave6 <- inner_join(df_female_domsp_weekday, relevant_wave6_data, by = "hicid") 
+df_females <- inner_join(domsp_wave6, relevant_wave8_data, by = "hicid") 
+#Take out missing data 
+df_females <- na.omit(df_females)
+```
+
+```{r}
+##Prepare mediator variables 
+#Select mediator variables (domain-specific movement behaviours)
+mediator_variables <- df_females %>% 
+  select(active_transport_at_10:nighttime_sleep_at_10)
+
+#Structure mediator variables to be a 2-dimensional matrix 
+mediator_list=as.list(mediator_variables)
+matrix(unlist(mediator_list), ncol = 13, byrow = FALSE) -> mediator_matrix
+#Updata column names to make sense
+colnames(mediator_matrix) <- c("v1_active_transport", "v2_naps", "v3_education", "v4_leisure_SB", "v5_passive_transport", "v6_screen_time", "v7_self_care", "v8_social", "v9_sports", "v10_unst_LPA", "v11_unst_MVPA", "v12_household", "v13_sleep")
+df_mediators <- as.data.frame(mediator_matrix)
+#Scale and center mediators - scale function centers by default 
+df_mediators <- scale(df_mediators) 
+```
+
+```{r}
+#Make interaction variables 
+for (x in 2:12){
+  assign(paste0("interactions_", x), (t(apply(df_mediators, 1, combn, x, prod))))
+  interaction_number <- get(paste0("interactions_",x))
+  colnames(interaction_number) <- paste(combn(1:13, x, paste, collapse="V"), sep = "_")
+  assign(paste0("interactions_", x), interaction_number)
+}
+interactions_13 <- df_mediators[,1]*df_mediators[,2]*df_mediators[,3]*df_mediators[,4]*df_mediators[,5]*df_mediators[,6]*df_mediators[,7]*df_mediators[,8]*df_mediators[,9]*df_mediators[,10]*df_mediators[,11]*df_mediators[,12]*df_mediators[,13]
+interactions_13 <- as.matrix(interactions_13)
+colnames(interactions_13) <- "1V2V3V4V5V6V7V8V9V10V11V12V13"
+mediators_interactions <- cbind(interactions_2, interactions_3, interactions_4, interactions_5, interactions_6, interactions_7, interactions_8, interactions_9, interactions_10, interactions_11, interactions_12, interactions_13) 
+``` 
+
+```{r}
+#Select exposure variables (SEP) - already standarized in LSAC
+exposure_variable <- df_females %>% 
+  select(SEP) 
+exposure_variable <- exposure_variable$SEP 
+```
+
+```{r}
+#Select and scale outcome variables 
+for (x in 123:ncol(df_females)) {
+  name <- colnames(df_females[,x]) 
+  assign(paste(name), df_females %>% 
+           select(x) %>%
+           scale())
+  assign(paste0(name),get(name)[,1])
+}
+
+#Make a dataframe of outcome variables 
+df_outcomes <- cbind(hapsoc, hahypr, haemot, hapeer,  hacondb, hasdqtb) 
+df_outcomes <- as.data.frame(df_outcomes)
+#Make lambda grid to be used in function to find best fit model 
+lambda.grid <- seq(from = 0.4, to = 0.01, by = -0.01)
```

Comment:
:bulb: see style guide for variable naming. Regardless, it's best not to mix naming conventions. Either use underscores or dots, but not both.